privacyidea/lib/tokens/pushtoken.py

```diff
@@ -189,6 +189,31 @@ def _build_smartphone_data(serial, challenge, registration_url, pem_privkey, opt
     message_on_mobile = get_action_values_from_options(SCOPE.AUTH,
                                                        PUSH_ACTION.MOBILE_TEXT,
                                                        options) or DEFAULT_MOBILE_TEXT
+    # Get the request object
+    req_headers = options.get("g", {}).get("request_headers", {})
+    req_environment = req_headers.environ
```

Comment:
In case there are no `request_headers` in `g`-object, this fails.
Maybe we should just set `request = None` in this case.